File: website/src/main/java/info/xiaomo/website/model/TechnologyModel.java
package info.xiaomo.website.model;


import info.xiaomo.core.base.BaseModel;
import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 * <p>
 * Date: 2016/11/3 14:27
 * Copyright(©) 2015 by xiaomo.
 **/

@Entity
@Table(name = "technology")
// lomlok
@Data
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = false)
@AllArgsConstructor
@NoArgsConstructor
public class TechnologyModel extends BaseModel {

    @Column(name = "Name")
    private String name;

    @Column(name = "Url")
    private String url;

    @Column(name = "Summary")
    private String summary;

    @Column(name = "ImgUrl")
    private String imgUrl;

}

File: website/src/main/java/info/xiaomo/website/model/TechnologyModel.kt
package info.xiaomo.website.model


import info.xiaomo.core.base.BaseModel
import lombok.EqualsAndHashCode
import lombok.NoArgsConstructor
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table

/**
 * 把今天最好的表现当作明天最新的起点．．～
 * いま 最高の表現 として 明日最新の始発．．～
 * Today the best performance  as tomorrow newest starter!
 * Created by IntelliJ IDEA.
 *
 * @author : xiaomo
 * github: https://github.com/xiaomoinfo
 * email: xiaomo@xiaomo.info
 *
 * Date: 2016/11/3 14:27
 * Copyright(©) 2015 by xiaomo.
 */

@Entity
@Table(name = "technology")
// lomlok

@EqualsAndHashCode(callSuper = false)

@NoArgsConstructor
class TechnologyModel : BaseModel() {

    @Column(name = "Url")
    var url: String? = null
        set(url) {
            field = this.url
        }

    @Column(name = "Summary")
    var summary: String? = null
        set(summary) {
            field = this.summary
        }

    @Column(name = "ImgUrl")
    var imgUrl: String? = null
        set(imgUrl) {
            field = this.imgUrl
        }

}
